source/javascripts/lib/wirsing/filters/search.coffee

```diff
@@ -0,0 +1,21 @@
+#= require ../module
+
+# Returns a configurable search function that can be used
+# to filter an array of objects by searching for a term
+# within the specified property of each object.
+
+wirsing.module 'filter.search', ->
+
+  (options) ->
+
+    search = (items) ->
+      if search.term.length < search.minChars then return items
+      else items.filter (item) -> item[search.in].search(search.term) > -1
+
+    search.configure = (config) ->
+      search.in = config.in || search.in || throw new Error 'option: <in> required'
+      search.term = config.term || search.term || ''
+      search.minChars = config.minChars || search.minChars || 3
```

Comment:
I could also use `or` here (actually i had it like that) but the pipes `||` make it clearer what's going on here (in my opinion). I would even argue that in this (special) case we could add that to our style guide.

The `||` is a one of the nicer hacks in Javascript where you can assign the first expression that evaluates to `true`, which is convenient in these default param situations. Technically we should use the existential operator `?` and this code is far from bullet-proof. But it would become more verbose and ugly :-D you can choose!

I found it strange to use `minChars` and not to expose it as configurable. But I can also hide it internally and fix it to 3 chars.